src/test/java/org/broadinstitute/hellbender/engine/filters/ReadFilterPluginUnitTest.java

```diff
@@ -348,50 +341,25 @@ public void testPreserveToolDefaultFilterOrder() {
         // Now get the final merged read filter and verify the execution order. We need to ensure that
         // getMergedReadFilter creates a composite filter that honors the filter test execution
         // order rules (tool defaults first, in order, followed by command line-specified, in order
-        // listed). So reach inside the filter and navigate down the tree. Since these are
-        // "and" filters that are built bottom-up, and we're traversing down, we visit the nodes
-        // in the opposite of the order their tests are executed.
+        // listed). So reach inside the filter and navigate down the tree.
         ReadFilter rf = instantiateFilter(clp, createHeaderWithReadGroups());
         String expectedOrder[] = {
-                // this list is in the order we encounter the nodes on the way down, which is the
-                // reverse of the order of test execution
-                ReadFilterLibrary.GOOD_CIGAR.getClass().getSimpleName(),
-                ReadFilterLibrary.HAS_MATCHING_BASES_AND_QUALS.getClass().getSimpleName(),
-                ReadFilterLibrary.NOT_SUPPLEMENTARY_ALIGNMENT.getClass().getSimpleName(),
-                ReadFilterLibrary.NOT_SECONDARY_ALIGNMENT.getClass().getSimpleName(),
-                ReadFilterLibrary.NOT_DUPLICATE.getClass().getSimpleName(),
-                ReadFilterLibrary.PROPERLY_PAIRED.getClass().getSimpleName(),
-                ReadFilterLibrary.FIRST_OF_PAIR.getClass().getSimpleName(),
-                ReadFilterLibrary.NONZERO_FRAGMENT_LENGTH_READ_FILTER.getClass().getSimpleName(),
-                ReadFilterLibrary.PAIRED.getClass().getSimpleName(),
-                ReadFilterLibrary.MAPPING_QUALITY_NOT_ZERO.getClass().getSimpleName(),
+                WellformedReadFilter.class.getSimpleName(),
                 ReadFilterLibrary.HAS_READ_GROUP.getClass().getSimpleName(),
-                WellformedReadFilter.class.getSimpleName()
+                ReadFilterLibrary.MAPPING_QUALITY_NOT_ZERO.getClass().getSimpleName(),
+                ReadFilterLibrary.PAIRED.getClass().getSimpleName(),
+                ReadFilterLibrary.NONZERO_FRAGMENT_LENGTH_READ_FILTER.getClass().getSimpleName(),
+                ReadFilterLibrary.FIRST_OF_PAIR.getClass().getSimpleName(),
+                ReadFilterLibrary.PROPERLY_PAIRED.getClass().getSimpleName(),
+                ReadFilterLibrary.NOT_DUPLICATE.getClass().getSimpleName(),
+                ReadFilterLibrary.NOT_SECONDARY_ALIGNMENT.getClass().getSimpleName(),
+                ReadFilterLibrary.NOT_SUPPLEMENTARY_ALIGNMENT.getClass().getSimpleName(),
+                ReadFilterLibrary.HAS_MATCHING_BASES_AND_QUALS.getClass().getSimpleName(),
+                ReadFilterLibrary.GOOD_CIGAR.getClass().getSimpleName()
```

Comment:
Yes. The order verifier used to validate the order in the reverse of the expected order. In this PR, I rewrote it to match reality so I could eliminate this confusion. You now list the expected values in the same order you expect the filter to test them.